src/components/layout/organize/SingleViewLayout.tsx

```diff
@@ -1,25 +1,56 @@
 import { Box } from '@material-ui/core';
 import { FunctionComponent } from 'react';
-import { useQuery } from 'react-query';
+import { useIntl } from 'react-intl';
 import { useRouter } from 'next/router';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
 
+import EditTextinPlace from 'components/EditTextInPlace';
 import getView from 'fetching/views/getView';
+import patchView from 'fetching/views/patchView';
 import TabbedLayout from './TabbedLayout';
 import ViewJumpMenu from 'components/views/ViewJumpMenu';
+import ZetkinQuery from 'components/ZetkinQuery';
+import { ZetkinView } from 'types/views';
 
 
 const SingleViewLayout: FunctionComponent = ({ children }) => {
+    const intl = useIntl();
     const router = useRouter();
+    const queryClient = useQueryClient();
     const { orgId, viewId } = router.query;
     const viewQuery = useQuery(['view', viewId ], getView(orgId as string, viewId as string));
+    const patchViewMutation = useMutation(patchView(orgId as string, viewId as string));
 
-    const view = viewQuery.data;
+    const updateTitle = async (view: ZetkinView, newTitle: string) => {
+        try {
+            await patchViewMutation.mutateAsync({ title: newTitle }, {
+                onSettled: () => queryClient.invalidateQueries(['view' ]),
```

Comment:
In `src/people/views/[viewId].tsx` that the query key used in `prefetchQuery` is `['view', viewId]`. Since that's the query we want to invalidate, we should use the same key. Just need to add `viewId` to the list!

Reply:
Also, I don't think we want to use try / catch here, and I don't think the return true / false is needed because it doesn't look like you're using the boolean state for anything. 

In the case you wanted to use these booleans to show an error message to the user if the handler threw an error, you would instead be better off using the `onSuccess` and `onError` properties in the same way you used `onSettled.

Reply:
Finally, this method doesn't need to be async. Since it's just going to resolve and invalidate a query. I would simply write it like this:

```js
const updateTitle = (view: ZetkinView, newTitle: string) => {
  patchViewMutation.mutate({ title: newTitle }, {
    onSettled: () => queryClient.invalidateQueries(['view' ]),
  })
};
```

Reply:
Yes, I realised that. But I think if this is a generic component that we would use everywhere, I don't think it's ideal to have to pass a function that returns a boolean in such a way, it enforces too much structure on the `onSubmit` handler, and doesn't use try/catch internally. I wonder if I'm using the the word "antipattern" correctly if I say that's what it looks like?